pkg/cache/multi_namespace_cache.go

```diff
@@ -0,0 +1,136 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	toolscache "k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type multiNamespaceCache struct {
+	namespaceToCache map[string]Cache
+}
+
+// NewMultiNamespacedCache - retrieve the new function for a multinamespaced cache.
+func NewMultiNamespacedCache(namespaces []string) func(config *rest.Config, opts Options) (Cache, error) {
```

Comment:
`NewMultiNamespaceCacheBuilder` or something, to imply that this doesn't return a multi-namespace-cache directly?